neurons/validator/__init__.py

```diff
@@ -158,23 +158,23 @@ def pull_task(self, synapse: PullTask) -> PullTask:
             bt.logging.debug(f"[{uid}] asked for a new task while having expired task. New task will be assigned")
             miner.reset_task()
 
-        cooldown_violation = True
         if miner.assigned_task is not None:
             bt.logging.debug(
-                f"[{uid}] asked for a new task while having assigned task. " f"Resetting the task and setting cooldown"
+                f"[{uid}] asked for a new task while having assigned task. Resetting the task and setting cooldown"
             )
-            cooldown_violation = False
             miner.reset_task(cooldown=self.config.generation.task_cooldown)
+            return self._add_cooldown_data(synapse, miner)
 
         if miner.is_on_cooldown():
-            if cooldown_violation:
-                miner.cooldown_violations += 1
+            miner.cooldown_violations += 1
             bt.logging.debug(
-                f"[{uid}] asked for a new task while on a cooldown. " f"Total violations: {miner.cooldown_violations}"
+                f"[{uid}] asked for a new task while on a cooldown. Total violations: {miner.cooldown_violations}"
             )
-            synapse.cooldown_until = miner.cooldown_until
-            synapse.cooldown_violations = miner.cooldown_violations
-            return synapse
+            if miner.cooldown_violations > self.config.neuron.cooldown_violation_threshold:
```

Comment:
this looks to complicated for a human eye
Is it possible to move it to variable?

Reply:
updated